crates/amaru-ledger/src/lib.rs

```diff
@@ -36,3 +36,43 @@ pub mod rules;
 pub mod state;
 pub mod store;
 pub mod summary;
+
+#[cfg(test)]
+pub(crate) mod test {
+    /// Creates a transaction input from a transaction hash str and an index.
+    ///
+    /// ## Required Imports
+    /// ```
+    /// use amaru_kernel::{Hash, TransactionInput};
+    /// ```
+    macro_rules! fake_input {
+        ($transaction_id:expr, $index:expr) => {
+            TransactionInput {
+                transaction_id: Hash::from(hex::decode($transaction_id).unwrap().as_slice()),
+                index: $index,
+            }
+        };
+    }
+
+    /// Creates a transaction output with no value, datum, or script ref with a specified address
+    ///
+    /// ## Required Imports
+    /// ```
+    ///     use amaru_kernel::{
+    ///    Bytes, PostAlonzoTransactionOutput, TransactionOutput, Value,
+    /// };
+    /// ```
+    macro_rules! fake_output {
+        ($address:expr) => {
+            TransactionOutput::PostAlonzo(PostAlonzoTransactionOutput {
+                address: Bytes::from(hex::decode($address).expect("Invalid hex address")),
+                value: Value::Coin(0),
+                datum_option: None,
+                script_ref: None,
+            })
+        };
+    }
```

Comment:
Not sure to see what benefits does the macro provides here over a function call 🤔 ?

Reply:
No real difference afaik, I just wanted to make it accessible from outside the `rules`, but I can just make it a public function.